elixir-move.el

```diff
@@ -0,0 +1,291 @@
+;;; elixir-move.el --- Navigate elixir source
+
+;; Copyright (C) 2015  Andreas Roehler
+
+;; Author: Andreas Roehler <andreas.roehler@online.de>
+;; Keywords: languages
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;;
+
+;;; Code:
+
+(defvar elixir-max-specpdl-size max-specpdl-size
+  "Protect against eternal loop")
+
+(defun elixir--skip-to-comment-or-semicolon ()
+  "Returns position if comment or semicolon found. "
+  (let ((orig (point)))
+    (cond ((and done (< 0 (abs (skip-chars-forward "^#;" (line-end-position))))
+		(member (char-after) (list ?# ?\,)))
+	   (when (eq ?\, (char-after))
+	     (skip-chars-forward ";" (line-end-position))))
+	  ((and (< 0 (abs (skip-chars-forward "^#;" (line-end-position))))
+		(member (char-after) (list ?# ?\,)))
+	   (when (eq ?\, (char-after))
+	     (skip-chars-forward ";" (line-end-position))))
+	  ((not done)
+	   (end-of-line)))
+    (skip-chars-backward " \t" (line-beginning-position))
+    (and (< orig (point))(setq done t)
+	 done)))
+
+(defun elixir--skip-to-semicolon-backward (&optional limit)
+  "Fetch the beginning of statement after a semicolon.
+
+Returns position reached if point was moved. "
+  (let ((orig (point)))
+    (and (< 0 (abs
+               (skip-chars-backward "^;" (or limit (line-beginning-position)))))
+         (skip-chars-forward " \t" (line-end-position))
+         (setq done t)
+         (and (< (point) orig) (point)))))
+
+(defalias 'elixir-backward-statement 'elixir-beginning-of-statement)
+(defalias 'elixir-previous-statement 'elixir-beginning-of-statement)
+(defalias 'elixir-statement-backward 'elixir-beginning-of-statement)
+(defun elixir-beginning-of-statement (&optional orig done limit)
+  "Go to the initial line of a simple statement.
+
+For beginning of compound statement use elixir-beginning-of-block.
+For beginning of clause elixir-beginning-of-clause.
+
+Referring python program structures see for example:
+http://docs.python.org/reference/compound_stmts.html"
+  (interactive)
+  (save-restriction
+    (unless (bobp)
+      (let* ((orig (or orig (point)))
+             (this (point))
+             (cui (current-indentation))
+
+             (pps (progn (goto-char this)
+                         (parse-partial-sexp (or limit (point-min))(point))))
+             (done done)
+             erg)
+        ;; lp:1382788
```

Comment:
On 17.03.2015 21:24, Matt DeBoard wrote:
>> +For beginning of clause elixir-beginning-of-clause.
>> +
>> +Referring python program structures see for example:
>> +http://docs.python.org/reference/compound_stmts.html"
>> +  (interactive)
>> +  (save-restriction
>> +    (unless (bobp)
>> +      (let* ((orig (or orig (point)))
>> +             (this (point))
>> +             (cui (current-indentation))
>> +
>> +             (pps (progn (goto-char this)
>> +                         (parse-partial-sexp (or limit (point-min))(point))))
>> +             (done done)
>> +             erg)
>> +        ;; lp:1382788
>
> What's this?
>

Will delete references to Python.
The clause itself seems to exist in Elixir.